metrics.go

```diff
@@ -0,0 +1,34 @@
+/*
```

Comment:
Do we want to put `metrics.go` and `profiling.go` at top level?

Reply:
If anyone else have an idea where is a suitable place for these files, then we can move them.